app/src/main/java/lv/spkc/apturicovid/ui/sms/SmsRetrievalFragment.kt

```diff
@@ -73,19 +73,17 @@ class SmsRetrievalFragment: BaseFragment() {
                 findNavController().popBackStack()
             }
 
-            resendCodeTv.setOnClickListener(object : DebouncedInfoClickListener(MILLIS_IN_MINUTE) {
-                override fun performClick(v: View?) {
-                    sendNumber()
-                    Toast.makeText(requireContext(), getString(R.string.label_code_resent), Toast.LENGTH_SHORT).show()
-                }
-
-                override fun onError(timeRemainingMillis: Long) {
-                    val errorText = String.format(getString(R.string.label_code_resend_error), timeRemainingMillis * -1 / MILLIS_IN_SECOND)
-                    Toast.makeText(requireContext(), errorText, Toast.LENGTH_SHORT).show()
-                }
-
-            })
+            resendCodeTv.setOnClickListener { smsViewModel.startCodeRetrievalTimer(MILLIS_IN_MINUTE, false) }
+        }
 
+        observeLiveData(smsViewModel.requestRemainingTimeShowMessagePairLiveData) {
+            if (it.first == 0L) {
+                sendNumber()
+                if (!it.second) Toast.makeText(requireContext(), getString(R.string.label_code_resent), Toast.LENGTH_SHORT).show()
+            } else {
+                val errorText = String.format(getString(R.string.label_code_resend_error), it.first * -1 / MILLIS_IN_SECOND)
```

Comment:
Kotlinā šis ir lieks -> "You entered: $stringInput" 

Ja ir garāks tad vēl ieerapi figūriekavās